examples/miscellaneous/plot_metadata_routing.py

```diff
@@ -169,7 +169,9 @@ def fit(self, X, y, **fit_params):
         # we can use provided utility methods to map the given metadata to what
         # is required by the underlying estimator. Here `method` refers to the
         # parent's method, i.e. `fit` in this example.
-        routed_params = request_router.route_params(params=fit_params, caller="fit")
+        routed_params = request_router.route_params(
+            params=fit_params, caller="fit", parent=self.__class__
```

Comment:
I'm confused here. I thought that `parent` should be `self` directly?

Reply:
Another thing that I'm realising now: `parent` different from `router.owner`? I'm under the impression that this the is same information and thus we would not need to have `parent`.

Reply:
I'm not sure about this. I will have to come back to this next week.